pychromecast/__init__.py

```diff
@@ -171,8 +178,23 @@ def __init__(self, host, port=None, device=None, **kwargs):
         self.status = None
         self.status_event = threading.Event()
 
+        self.browser = None
+        self.tracker = None
+        self._socket_connection_status = None
+
+        self.setup(host, port, self.device, self.kwargs)
+
+    def setup(self, host, port, device, kwargs):
+        """
+        Method for setting up, and resetting the chromecast object
+        """
+        tries = kwargs.pop('tries', None)
```

Comment:
Another approach would be to default to a certain number (2?) of tries for a cast group and only listen to DISCONNECTED events. Might be a cleaner solution actually. What do you think?